R/epidist.R

```diff
@@ -2,73 +2,121 @@
 #'
 #' Parameter probability distribution by day
 #' @param pathogen A \code{character} specifying pathogen of interest
-#' @param pathogen A \code{character} specifying type of parameter: "incubation", "onset_to_admission", "onset_to_death"
-#' @param study A \code{character} specifying dataset to use. Defaults to study with largest sample size.
+#' @param type A \code{character} specifying type of parameter: "incubation",
+#' "onset_to_admission", "onset_to_death"
+#' @param study A \code{character} specifying dataset to use. Defaults to study
+#' with largest sample size.
 #' @keywords incubation
 #' @export
 #' @examples
-#' epidist()
+#' # list_distributions() will show which pathogens are available for each
+#' # metric here we search for which incubation periods are available
+#' list_distributions(type = "incubation")
+#'
+#' # example of epidist() using incubation period for ebola
+#' epidist(pathogen = "ebola", type = "incubation")
+#'
+#' # when more than one study is available in the database a study can be
+#' # specified
+#' epidist(
+#'   pathogen = "MERS_CoV",
+#'   type = "incubation",
+#'   study = "Cauchemez_et_al"
+#' )
+#'
+#' # example using onset to admission as the metric
+#' epidist(pathogen = "ebola", type = "onset_to_admission")
+epidist  <- function(pathogen, type, study = NULL) {
 
-epidist  <- function(pathogen, type, study = NULL){
-  
   # DEBUG
   # pathogen="SARS1"; type="incubation"; study=NULL; pmf = T
-  
+
+  pathogen_ID <- NULL # remove global variable note
+  type_ID <- NULL # remove global variable note
+  size <- NULL # remove global variable note
+  study_ID <- NULL # remove global variable note
+
   # Extract pathogen and parameter type
-  pick_path <- epiparameter:::param_vals %>% dplyr::filter(pathogen_ID==pathogen & type_ID==type)  
-  
-  if(nrow(pick_path)==0){stop("Need to select pathogen and distribution in the dataset")}
-  
+  pick_path <- utils::read.csv(file = system.file(
+    "extdata",
+    "parameters.csv",
+    package = "epiparameter",
+    mustWork = TRUE
+    )) %>% dplyr::filter(pathogen_ID == pathogen & type_ID == type)
+
+  if (nrow(pick_path) == 0) {
+    stop("Need to select pathogen and distribution in the dataset")
+  }
+
   # Extract study or default to largest sample size
-  if( is.null(study) ){pick_study <- pick_path %>% dplyr::filter(size==max(size))}
-  if(!is.null(study) ){pick_study <- pick_path %>% dplyr::filter(study_ID==study)}
+  if (is.null(study)) {
+    pick_study <- pick_path %>% dplyr::filter(size == max(size))
+  }
+  if (!is.null(study)) {
+    pick_study <- pick_path %>% dplyr::filter(study_ID == study)
+  }
```

Comment:
I would remove all instances of the piping operator. They make debugging really hard, I think the general recommendation is not to use them within packages.

Reply:
I agree, I had planned to do it after setting up the package, but more than happy to include it in this PR. I have now replaced the use of the tidyverse pipe with the base pipe, and deleted the import of the magrittr pipe. If you think that removal of the the base pipe is also recommended I am happy to go with that instead. What do you think?